griptape/drivers/memory/conversation/griptape_cloud_conversation_memory_driver.py

```diff
@@ -126,15 +126,18 @@ def load(self) -> tuple[list[Run], dict[str, Any]]:
         # retrieve the Thread to get the metadata
         thread_response = self._call_api("get", f"/threads/{thread_id}").json()
 
-        runs = [
-            Run(
-                id=m["metadata"].pop("run_id"),
+        runs = []
+        for m in messages_response.get("messages", []):
+            run_id = m["metadata"].pop("run_id", None)
+            run = Run(
                 meta=m["metadata"],
                 input=BaseArtifact.from_json(m["input"]),
                 output=BaseArtifact.from_json(m["output"]),
             )
-            for m in messages_response.get("messages", [])
-        ]
+            if run_id is not None:
+                run.id = run_id
```

Comment:
There could be subtle issues with assigning `id` after creation (future `attrs_post_init` logic). We should still do it at time of instantiation. Either with something like:

```python
        runs = [
            Run(
                **({"id": message["metadata"]} if "metadata" in message else {}),
                meta=message["metadata"],
                input=BaseArtifact.from_json(message["input"]),
                output=BaseArtifact.from_json(message["output"]),
            )
            for message in messages_response.get("messages", [])
        ]
```
Or by separating the param creation and run creation into two separate steps.

Reply:
updated